modules/packages/Python.chpl

```diff
@@ -653,9 +730,33 @@ module Python {
     /*
       Converts a python dictionary to an associative array
     */
-    // proc fromDict(type T, obj: PyObjectPtr): T throws {
-      // TODO
-    // }
+    proc fromDict(type T, obj: PyObjectPtr): T throws
+      where isArrayType(T) {
+
+      // no init causes segfaults :(
+      var dummy: T;
+
+      // rebuild the array with a modifiable domain
+      var dom = dummy.domain;
```

Comment:
Isn't there a standard library function that you can use to get stuff out of arrays like this, without creating a dummy? I am lost in the `chpl__whatever` functions that can help, but Vass would know, I think.

Reply:
I am not sure. @vasslitvinov, do you know a better way to do this?

Reply:
```chpl
__primitive("get runtime type field", rtt, "dom")
// for example, in our internal modules:
const ref dom = chpl__domainFromArrayRuntimeType(dstType);
```